clearnode/docs/SessionKeys.md

```diff
@@ -0,0 +1,238 @@
+#### Special Application: "clearnode"
+
+Session keys registered with the application name `"clearnode"` receive special treatment:
+
+- **Root Access**: These session keys bypass spending allowance validation and application restrictions
+- **Full Permissions**: They can perform any operation the wallet itself could perform
+- **Backward Compatibility**: This special behavior facilitates migration from older versions
+- **Expiration Still Applies**: Even with root access, the session key expires according to its `expires_at` timestamp
+
+> **Note:** The "clearnode" application name is primarily for backward compatibility and will be deprecated after a migration period for developers.
+
+### Expiration
+
+All session keys must have an **expiration timestamp** (`expires_at`) that defines when the session key becomes invalid:
+
+- **Future Timestamp Required**: The expiration time must be set to a future date when registering a session key
+- **Automatic Invalidation**: Once the expiration time passes, the session key can no longer be used for any operations
+- **Grace Period**: After expiration, the session key is removed from the active keys list
+- **Applies to All Keys**: Even "clearnode" application session keys must respect the expiration timestamp
+
+### Allowances
+
+Allowances define **spending limits** for session keys, specifying which assets the session key can spend and how much:
+
+```json
+{
+  "allowances": [
+    {
+      "asset": "usdc",
+      "amount": "100.0"
+    },
+    {
+      "asset": "eth",
+      "amount": "0.5"
+    }
+  ]
+}
+```
+
+#### Allowance Validation
+
+- **Supported Assets Only**: All assets specified in allowances must be supported by the system. Unsupported assets cause authentication to fail
+- **Usage Tracking**: The system tracks spending per session key by recording which session key was used for each ledger debit operation
+- **Spending Limits**: Once a session key reaches its spending cap for an asset, further operations requiring that asset are rejected with: `"operation denied: insufficient session key allowance: X required, Y available"`
+- **Empty Allowances**: Providing an empty `allowances` array (`[]`) means zero spending allowed for all assets—any operation attempting to spend funds will be rejected
+
+> **Important:** Only one session key is allowed per wallet+application combination. If you register a new session key for the same application, the old one is automatically invalidated and removed from the database.
+
+#### Allowances for "clearnode" Application
+
+Session keys with `application: "clearnode"` are exempt from allowance enforcement:
+
+- **No Spending Limits**: Allowance checks are bypassed entirely
+- **Full Financial Access**: These keys can spend any amount of any supported asset
+- **Expiration Still Matters**: Even without allowance restrictions, the session key still expires according to its `expires_at` timestamp
+
+## How to Manage Session Keys
+
+### Create and Configure
```

Comment:
Could you extract this to "Clearnode", and add a separate "Nitrolite SDK" section, please?